pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -251,7 +253,7 @@ int updatePermitsAndPendingAcks(final List<Entry> entries) throws PulsarServerEx
             }
             if (pendingAcks != null) {
                 PositionImpl pos = (PositionImpl) entry.getPosition();
-                pendingAcks.put(pos, batchSize);
+                pendingAcks.put(pos.getLedgerId(), pos.getEntryId(), batchSize, 0);
```

Comment:
One possible improvement here would be to skip the `PositionImpl` all together. Expose `getLedgerId()` and `getEntryId()` on the `EntryImpl` and do like: 

```java
if (pendingAcks != null) {
    EntryImpl e = (EntryImpl) entry;
    pendingAcks.put(e.getLedgerId(), e.getEntryId(), batchSize, 0);
}
```